lib/vsc/filesystem/posix.py

```diff
@@ -558,3 +559,66 @@ def create_stat_directory(self, path, permissions, uid, gid, override_permission
             self.log.debug("Path %s already exists with correct ownership", path)
 
         return created
+
+    def replace_acl(self, path, permissions):
+        """
+        Overwrite ACL permissions on given path
+        """
+        path = self._sanity_check(path)
+        fs = self._what_filesystem(path)
+        if fs is None:
+            self.log.warning(f"ACL replacement requested on  filesystem with unknown support: {path}")
+            return False
+
+        if not isinstance(permissions, (tuple, list,)):
+            raise PosixOperationError("Given ACL permissions are not a list or tuple")
+
+        replace_acl_method = self._replace_acl_posix
+        if fs[0] in ['gpfs', 'nfs4']:
+            replace_acl_method = self._replace_acl_nfs
+
+        return replace_acl_method(path, permissions)
+
+    def _replace_acl_posix(self, path, permissions):
+        """
+        Overwrite ACL permissions in POSIX format on given path
+        """
+        setfacl_exe = 'setfacl'
+
+        with tempfile.NamedTemporaryFile() as acl_file:
+            self.log.debug(f"Input file for {setfacl_exe} created in: {acl_file.name}")
+            acl_file.write('\n'.join(permissions).encode('utf-8'))
```

Comment:
can you also write to debug what the contents is? Now it's not visible what exactly will be done.